fido2.go

```diff
@@ -1,6 +1,13 @@
 package libfido2
 
 /*
+#cgo darwin CFLAGS: -I/usr/local/opt/openssl@1.1/include
+#cgo darwin,!libfido2static LDFLAGS: -lfido2
+#cgo darwin,libfido2static LDFLAGS: -framework CoreFoundation -framework IOKit /usr/local/lib/libfido2.a /usr/local/lib/libcbor.a /usr/local/opt/openssl@1.1/lib/libcrypto.a
```

Comment:
This is what I got, assuming you'd want to install `libcbor.a` in `/opt/homebrew` on ARM64 macOS:
```suggestion
#cgo darwin,amd64 CFLAGS: -I/usr/local/opt/openssl@1.1/include
#cgo darwin,amd64,!libfido2static LDFLAGS: -lfido2
#cgo darwin,amd64,libfido2static LDFLAGS: -framework CoreFoundation -framework IOKit /usr/local/lib/libfido2.a /usr/local/lib/libcbor.a /usr/local/opt/openssl@1.1/lib/libcrypto.a

#cgo darwin,arm64 CFLAGS: -I/opt/homebrew/opt/openssl@1.1/include -I/opt/homebrew/opt/libfido2/include
#cgo darwin,arm64,!libfido2static LDFLAGS: /opt/homebrew/opt/libfido2/lib/libfido2.dylib
#cgo darwin,arm64,libfido2static LDFLAGS: -framework CoreFoundation -framework IOKit /opt/homebrew/opt/libfido2/lib/libfido2.a /opt/homebrew/lib/libcbor.a /opt/homebrew/opt/openssl@1.1/lib/libcrypto.a
```

I'm still of the opinion that we could ditch nonstatic builds altogether and just ship our static libraries in the repo, it seems to be a somewhat common thing to do for go packages, and it was what the upstream lib did.

Reply:
Not sure what to do about libcrypto in that case, tho 😞

Reply:
> assuming you'd want to install libcbor.a in /opt/homebrew/lib on ARM64 macOS

I haven't tried arm64 builds yet, but I suppose not? I'll build both libfido2 and libcbor from sources on arm too, so we can guarantee it uses the same versions as everything else - that probably means pulling both from /usr/local/lib.

Dynamic builds could get away with only `-L/opt/homebrew/opt/libfido2/lib -lfido2`, I think. Doesn't homebrew have a general /lib folder on arm, akin to /usr/local/lib? /opt/homebrew/lib or something like that?

> I'm still of the opinion that we could ditch nonstatic builds altogether and just ship our static libraries in the repo

I could pull the entire build matrix here, but it feels like we are pulling complexity from Teleport into the wrong repo. I'd rather just point to a few (fixed) locations and give library users the flexibility to build dependencies however they like. If the upstream took that stance I wouldn't need to fork. We aren't in a position to just pull whatever compiled lib they embed, as I can't verify how it was built.